packages/pwa/app/commerce-api/hooks/useCustomerProductList.js

```diff
@@ -0,0 +1,325 @@
+    const api = useCommerceAPI()
+    const customer = useCustomer()
+    const _super = useCustomerProductLists()
+    const {actions} = useContext(CustomerProductListsContext)
+    const DEFAULT_LIST_TYPE = 'wish_list'
+    const type = options?.type || DEFAULT_LIST_TYPE
+
+    const self = useMemo(() => {
+        return {
+            isInitialized: _super.isInitialized,
+            reset: _super.reset,
+
+            get data() {
+                return Object.values(_super.data).find((list) => list.name === name)
+            },
+
+            get isEmpty() {
+                return !self.data?.customerProductListItems?.length
+            },
+
+            findItemByProductId(productId) {
+                return self.data?.customerProductListItems?.find(
+                    (item) => item.productId === productId
+                )
+            },
+
+            isProductInList(productId) {
+                return !!self.findItemByProductId(productId)
+            },
+
+            /**
+             * Initialize customer's product list.
+             */
+            async init() {
+                actions.setInitialized(false)
+                const list = await self._getOrCreatelistByName(name, {type})
+                const productDetails = await self._getProductDetails(list)
+                const result = self._mergeProductDetailsIntoList(list, productDetails)
+                actions.receiveList(result)
+                actions.setInitialized(true)
+                return list
+            },
+
+            async getList(listId) {
+                const response = await api.shopperCustomers.getCustomerProductList({
+                    parameters: {
+                        customerId: customer.customerId,
+                        listId
+                    }
+                })
+
+                if (isError(response)) {
+                    throw new Error(response)
+                }
+
+                return response
+            },
+
+            async addItem(product) {
+                let list = self.data
+                if (!self.isInitialized) {
+                    list = await self.init()
+                }
+                const createdItem = await self.createListItem(list.id, product)
+                actions.createListItem(list.id, createdItem)
+            },
+
+            async updateItem(item) {
+                const {id, quantity} = item
+                if (quantity === 0) {
+                    await self.removeListItem(self.data.id, id)
+                    actions.removeListItem(self.data.id, id)
+                    return
+                }
+                const updatedItem = await self.updateListItem(self.data.id, item)
+                actions.updateListItem(self.data.id, updatedItem)
+            },
+
+            async removeItem(itemId) {
+                await self.removeListItem(self.data.id, itemId)
+                actions.removeListItem(self.data.id, itemId)
+            },
```

Comment:
What is the reasoning for splitting the logic of addItem/createListItem, updateItem/updateListItem, and removeItem/removeListItem? 

If there wasn't the special logic for lazy creating and removal on 0, all 3 functions would be 2 liners. To me, createListItem/updateListItem/removeListItem sound like functions that could be defined in the `useCustomerProductLists` hook.

Example:
```
            async addItem(product) {
                let list = self.data
                if (!self.isInitialized) {
                    list = await self.init()
                }
                await customerProductLists.createListItem(list.id, product)
            },

            async updateItem(item) {
                const {id, quantity} = item
                if (quantity === 0) {
                    await customerProductLists.removeListItem(self.data.id, id)
                    return
                }
                await customerProductLists.updateListItem(self.data.id, item)
            },
            
            async removeItem(itemId) {
                await customerProductLists.removeListItem(self.data.id, itemId)
            }
```

To me I feel like the heavy lifting should be done in the `useCustomerProductLists` hook and not in the `useCustomerProductList` hook. I think a supporting point is that you are never going to use the function `createListItem` after instantiating a `useCustomerProductList` hook, because the listId will always be the same, there shouldn't be a function that allows you to modify other lists.